frontend/src/components/chats/components/current-chat/components/messages-list/messages-list.tsx

```diff
@@ -11,21 +11,49 @@ type Props = {
 };
 
 const MessagesList: FC<Props> = ({ messages, currentUserId }) => {
+  const msgs: JSX.Element[] = [];
+  const filteredMsgs: JSX.Element[] = [];
+
+  messages.map((message) => {
+    msgs.push(
+      <Message
+        messageAuthor={
+          message.sender.id === currentUserId ? 'user' : 'opponent'
+        }
+        content={message.message}
+        postTime={getFormattedDate(message.createdAt, 'HH:mm')}
+        originalPostTime={getFormattedDate(message.createdAt, 'yyyy-MM-dd')}
+        messageAvatarUrl={
+          message.sender.userDetails.avatar?.url ?? defaultAvatar
+        }
+      />,
+    );
+  });
+
+  msgs.reverse().reduce((prev, current) => {
+    if (prev.props.originalPostTime !== current.props.originalPostTime) {
+      filteredMsgs.push(
+        <DateSeparator postTime={prev.props.originalPostTime} />,
+      );
+    }
+    filteredMsgs.push(current);
+
+    return current;
+  });
+
+  const messagesWithSeparators = filteredMsgs.reverse();
```

Comment:
You're reversing messages array before reduce and then reverse it back - I don't think that it's necessary